File: JINSEOP/프로그래머스_WEEK12/거스름돈.cpp
#include <string>
#include <vector>
using namespace std;

int DP[100001];
int solution(int n, vector<int> money) {
    int answer = 0;
    DP[0] = 1;

    for (int i = 0; i < money.size(); i++) {
        for (int j = money[i]; j <= n; j++) {
            DP[j] += DP[j - money[i]];
            DP[j] %= 1000000007;
        }
    }

    answer = DP[n];
    return answer;
}

File: DONGUK/프로그래머스_WEEK_12/거스름돈.py
def solution(n, money):
    answer = 0
    
    dp = [0] * (n+1)
    
    dp[0] = 1
    for coin in money:
        for num in range(coin, n+1):
            dp[num] += dp[num - coin]
        
        
    answer = dp[n]
    
    return answer